waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -32,6 +32,7 @@ when defined(rlnzerokit):
 
 
 type MerkleNodeResult* = Result[MerkleNode, string]
+type BoolResult* = Result[bool, string]
 type RateLimitProofResult* = Result[RateLimitProof, string]
```

Comment:
A little nit-picking, but I think that having all these result types can be improved. And more if they are not going to evolve separately.

I suggest doing the following:
* Have a generic base result type alias for the RLN relay:
```nim
type RlnRelayResult*[T] = Result[T, string]
```
And then define special result types aliases, only in the case they are used in more than one place:

```nim
type MerkleNodeResult* = RlnRelayResult[MerkleNode]
type RateLimitProofResult* = RlnRelayResult[RateLimitProof]
```
> Disclaimer: I have not checked the number of usages of the following type aliases. If they have been used only in one place, then it is preferable to go with the generic type directly.

I do not like the `BoolResult` 🤔; with my proposal, the result of the method would become:
```nim
proc foo*(): Future[RlnRelayResult[bool]] =
  # [...]
```

Reply:
Addressed in [87661ba](https://github.com/status-im/nwaku/pull/1158/commits/87661ba1d572a90bbba441fca568e2dad9fb5fa9)

We will probably update this in one of the refactoring PRs to include a second generic type, the Error enum